pytket/qir/conversion/conversion.py

```diff
@@ -178,14 +178,54 @@ def __init__(
             ),
         )
 
-        # void __quantum__rt__int_record_output(i64)
-        self.record_output = self.module.module.add_external_function(
-            "__quantum__rt__int_record_output",
-            pyqir.FunctionType(
-                pyqir.Type.void(self.module.module.context),
-                [pyqir.IntType(self.module.module.context, qir_int_type)],
-            ),
-        )
+        self.reg_const = {}
+
+        for creg in self.circuit.c_registers:
+            reg_name = creg[0].reg_name
+            self.reg_const[reg_name] = self.module.module.add_byte_string(
+                str.encode(reg_name)
+            )
+
+        if len(self.reg_const) > 0:
+
+            # void __quantum__rt__int_record_output(i64)
+            self.record_output_i64 = self.module.module.add_external_function(
+                "__quantum__rt__int_record_output",
+                pyqir.FunctionType(
+                    pyqir.Type.void(self.module.module.context),
+                    [
+                        pyqir.IntType(self.module.module.context, qir_int_type),
+                        self.reg_const[list(self.reg_const)[0]].type,
```

Comment:
What's special about the zeroth member of `list(self.reg_const)`? What if there are multiple?

Reply:
From my understanding there is nothing special about the first element, they all should have the same type. The i8 pointer type is not directly available so I have to extract it from the given value.

Reply:
Yes, good point. Done in @cqc-melf
[add point type](https://github.com/CQCL/pytket-qir/pull/38/commits/3439488874d80935be430393f16f0ac0aae1dc4a)
[3439488](https://github.com/CQCL/pytket-qir/pull/38/commits/3439488874d80935be430393f16f0ac0aae1dc4a)